docs/workflows/create_workflow_block.md

```diff
@@ -0,0 +1,1898 @@
+        engine will understand that the step returns batch of elements for each input element and
+        create nested sturcures of indices to keep track of during execution of downstream steps.
+
+    === "output dimensionality decrease"
+      
+        In this example, the block visualises crops predictions and create tiles
+        presenting all crops predictions in single output image.
+
+        ```{ .py linenums="1" hl_lines="31-33 50-51 61-62"}
+        from typing import List, Literal, Type, Union
+
+        import supervision as sv
+        
+        from inference.core.workflows.execution_engine.entities.base import (
+            Batch,
+            OutputDefinition,
+            WorkflowImageData,
+        )
+        from inference.core.workflows.execution_engine.entities.types import (
+            BATCH_OF_IMAGES_KIND,
+            BATCH_OF_OBJECT_DETECTION_PREDICTION_KIND,
+            StepOutputImageSelector,
+            StepOutputSelector,
+            WorkflowImageSelector,
+        )
+        from inference.core.workflows.prototypes.block import (
+            BlockResult,
+            WorkflowBlock,
+            WorkflowBlockManifest,
+        )
+        
+        
+        class BlockManifest(WorkflowBlockManifest):
+            type: Literal["my_plugin/tile_detections@v1"]
+            crops: Union[WorkflowImageSelector, StepOutputImageSelector]
+            crops_predictions: StepOutputSelector(
+                kind=[BATCH_OF_OBJECT_DETECTION_PREDICTION_KIND]
+            )
+        
+            @classmethod
+            def get_output_dimensionality_offset(cls) -> int:
+                return -1
+        
+            @classmethod
+            def describe_outputs(cls) -> List[OutputDefinition]:
+                return [
+                    OutputDefinition(name="visualisations", kind=[BATCH_OF_IMAGES_KIND]),
+                ]
+        
+        
+        class TileDetectionsBlock(WorkflowBlock):
+        
+            @classmethod
+            def get_manifest(cls) -> Type[WorkflowBlockManifest]:
+                return BlockManifest
+        
+            def run(
+                self,
+                crops: Batch[WorkflowImageData],
+                crops_predictions: Batch[sv.Detections],
+            ) -> BlockResult:
+                annotator = sv.BoundingBoxAnnotator()
+                visualisations = []
+                for image, prediction in zip(crops, crops_predictions):
+                    annotated_image = annotator.annotate(
+                        image.numpy_image.copy(),
+                        prediction,
+                    )
+                    visualisations.append(annotated_image)
+                tile = sv.create_tiles(visualisations)
+                return {"visualisations": tile}
+        ```
+
+        * in lines `31-33` manifest class declares output dimensionality 
+        offset - value `-1` should be understand as decreasing dimensionality level by `1`
```

Comment:
`should be understand` -> `should be understood`

Reply:
👍